scripts/ci/validate-metadata-icons.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* eslint-env node */
+
+'use strict';
+
+const fs = require( 'fs-extra' );
+const glob = require( 'fast-glob' );
+const upath = require( 'upath' );
+const { table, getBorderCharacters } = require( 'table' );
+const { red, magenta } = require( './ansi-colors' );
+
+module.exports = async function validateMetadataIcons( { cwd = process.cwd() } = {} ) {
+	console.log( magenta( 'Validating metadata files.' ) );
+
+	const globPattern = upath.join( cwd, 'packages', '*', 'ckeditor5-metadata.json' );
+	const metadataFilePaths = await glob( globPattern );
+	const plugins = ( await Promise.all( metadataFilePaths.map( getPluginsFromMetadataFile ) ) ).flat();
+	const pluginIcons = plugins.reduce( getPluginIconPaths, {} );
+
+	const missingIcons = [];
+
+	for ( const packageName in pluginIcons ) {
+		for ( const iconSource of pluginIcons[ packageName ] ) {
+			let missing = false;
+
+			if ( iconSource.startsWith( '@ckeditor/' ) ) {
+				try {
+					require.resolve( iconSource );
+				} catch ( err ) {
+					missing = true;
+				}
+			} else {
+				const iconPath = upath.join( cwd, 'packages', packageName, iconSource );
+				missing = !( await fs.exists( iconPath ) );
+			}
+
+			if ( missing ) {
+				missingIcons.push( [ `./packages/${ packageName }/ckeditor5-metadata.json`, iconSource ] );
+			}
+		}
+	}
+
+	if ( !missingIcons.length ) {
+		return;
+	}
+
+	console.log( red( 'Detected invalid paths. Check the following paths in following packages:' ) );
```

Comment:
"Detected invalid paths" - what paths? The message sent before ("Validating metadata files.") also does not suggest to someone who sees such errors for the first time what paths it is about.

```suggestion
	console.log( red( 'Detected invalid icon paths in the following packages:' ) );
```